libraries/AP_HAL/Device.h

```diff
@@ -26,13 +26,24 @@
  */
 class AP_HAL::Device {
 public:
+    enum BusType {
+        I2C,
+        SPI,
```

Comment:
either use `enum class` or give a better name to the constants: BUS_TYPE_I2C, BUS_TYPE_SPI. I'm ok with the second approach since we already use SPEED_HIGH below